### PR TITLE
Daisy/games scraper

### PR DESCRIPTION
**Pull request by daisychangm**
## Overview

I created a web scraper for the schedules of each sport and updated the game and team fields.

## Changes Made

I created a web scraper with BeautifulSoup to scrape each Cornell sports schedule page and store each game/team. A game is updated if it already exists in the database and has a different result than the one stored. A new team is created if the opposing team does not exist in the database. 

## Test Coverage

I manually tested this feature using Postman.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,25 @@
-# score-backend
\ No newline at end of file
+# score-backend
+
+Tech stack:
+
+1. Typescript/Node.js
```
Comment (Aayush-Agnihotri) [inline README.md:5]: wait u guys r using python lmao

File: src/database.py [hunk 2]
```diff
@@ -6,4 +6,11 @@
 
 
 def init_db():
+    """
+    Initializes the database connection.
+
+    This function should be called once at the start of the application and will create the
+    database and collections if they don't exist.
+    """
+    print(db.list_collection_names())
```
Comment (Aayush-Agnihotri) [inline src/database.py:15]: not sure if initializing the client above does what this function is meant to do, so we can probably just delete it

File: src/mutations/create_game.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        date,
+        gender,
+        opponent_id,
+        sport,
+        state,
+        location=None,
+        result=None,
+        time=None,
+    ):
+        print("A")
```
Comment (Aayush-Agnihotri) [inline src/mutations/create_game.py:33]: delete the print

File: src/mutations/create_game.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            "date": date,
+            "gender": gender,
+            "location": location,
+            "opponent_id": opponent_id,
+            "result": result,
+            "sport": sport,
+            "state": state,
+            "time": time,
+        }
+        print(game_data["city"])
```
Comment (Aayush-Agnihotri) [inline src/mutations/create_game.py:45]: delete

File: src/schema.py [hunk 5]
```diff
@@ -1,57 +1,15 @@
...
-                city=new_game.city,
-                date=new_game.date,
-                gender=new_game.gender,
-                opponent_id=new_game.opponent_id,
-                sport=new_game.sport,
-                state=new_game.state,
-                time=new_game.time,
-            )
-        )
+class Query(TeamQuery, GameQuery, ObjectType):
```
Comment (Aayush-Agnihotri) [inline src/schema.py:6]: Add the queries here

File: src/scrapers/games_scraper.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        "result": game_data["result"],
+        "sport": game_data["sport"],
+        "state": state,
+        "time": game_data["time"],
+    }
+
+    GameService.create_game(game_data)
+
+
+# if __name__ == "__main__":
```
Comment (Aayush-Agnihotri) [inline src/scrapers/games_scraper.py:101]: delete this comments to keep the code clean

File: src/services/game_service.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        return GameRepository.find_by_id(game_id)
+
+    @staticmethod
+    def create_game(data):
+        """
+        Create a new game.
+        """
+        opponent_id = data.get("opponent_id")
+        if not TeamService.get_team_by_id(opponent_id):
+            print(f"Opponent team with id {opponent_id} does not exist.")
```
Comment (Aayush-Agnihotri) [inline src/services/game_service.py:28]: we can delete the print if we are also raising an error

File: src/utils/helpers.py [hunk 8]
```diff
@@ -29,5 +41,8 @@ def get_dominant_color(relative_path, white_threshold=200, black_threshold=50):
...
     return dominant_color
 
-dominant_color = get_dominant_color('/images/logos/Princeton_Tigers.png?width=80&height=80&mode=max')
-print(f"Dominant color: {dominant_color}")
\ No newline at end of file
+
+dominant_color = get_dominant_color(
+    "/images/logos/Princeton_Tigers.png?width=80&height=80&mode=max"
+)
+print(f"Dominant color: {dominant_color}")
```
Comment (Aayush-Agnihotri) [inline src/utils/helpers.py:48]: delete the test and the print statement

File: src/utils/helpers.py [hunk 9]
```diff
@@ -3,22 +3,34 @@
 from io import BytesIO
 from collections import Counter
 
-BASE_URL = "https://dxbhsrqyrr690.cloudfront.net/sidearm.nextgen.sites/cornellbigred.com"
+BASE_URL = (
```
Comment (Aayush-Agnihotri) [inline src/utils/helpers.py:6]: move this into the constants.py file as like IMAGE_BASE_URL or something

**Review by Aayush-Agnihotri — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (daisychangm) [reply README.md:5]: Oh oops I copied it from volume I think I forgot to change it

**Review by TushigBili — APPROVED** [review]
Looks good!

**Pull request merged**